CMakeLists.txt

```
@@ -120,14 +120,19 @@ else ()
 	list(APPEND libobjc_C_SRCS eh_personality.c)
 endif (WIN32)
 
-if (NOT EXISTS "${CMAKE_SOURCE_DIR}/third_party/robin-map/include/tsl/robin_map.h")
-	message(FATAL_ERROR "Git submodules not present, please run:\n\n"
-						" $ git submodule init && git submodule update\n\n"
-						"If you did not checkout via git, you will need to"
-						"fetch the submodule's contents from"
-						"https://github.com/Tessil/robin-map/")
-endif ()
+find_path(ROBIN_MAP_HEADER NAMES "tsl/robin_map.h")
```

```suggestion
find_package(tsl-robin-map)
```

It should come with a CMake package. You'll also need to do something with the found file, right? I think ROBIN_MAP_HEADER is unused right now except for checking if it's there.

```cmake
target_link_libraries(objc PRIVATE tsl::robin_map)
```

---

Correct, that works because the headers are installed into `/include/`.  I've added:
```cmake
include_directories("${tsl-robin-map_INCLUDE_DIRS}")
```
but it's redundant in most cases.

The package is a header-only package; there is no library to link against.

---

CMake models header-only libraries just like other libraries. If you import the target then it adds the relevant flags to find the headers. It also adds any compile flags that the ‘library’ needs.